go/registry/api/runtime.go

```diff
@@ -160,6 +160,13 @@ type Runtime struct {
 
 	// Storage stores parameters of the storage committee.
 	Storage StorageParameters `json:"storage,omitempty"`
+
+	// NodeEntityWhitelist is a map of entity IDs that may register nodes for this runtime to the true.
+	// This list applies to all roles.
+	NodeEntityWhitelist map[signature.PublicKey]bool `json:"node_entity_whitelist"`
+
+	// AllowAnyNode--if this is set, then nodes from entities other than those in NodeEntityWhitelist are allowed too.
+	AllowAnyNode bool `json:"allow_any_node"`
```

Comment:
Instead of this I would have an `AdmissionPolicyKind` enum with `AdmissionPolicyEntityWhitelist` and `AdmissionPolicyAnyone` kinds. (Default kind should be invalid and rejected, as per #2527).

So maybe it would make sense to make a separate `AdmissionPolicy` structure (and corresponding field in `Runtime`) with `Kind` and `EntityWhitelist` fields to group things nicely and make it less messier to extend in the future.

Reply:
our other tagged union hack was to have a struct with a bunch of pointer fields

Reply:
hm although that was probably for rust interop

Reply:
Thought about that as well, I would be ok with that too.

Reply:
oh, one of the variants is a unit type, so there will be no rust interop today

Reply:
I went with pointer fields \: